egui_demo_lib/src/apps/demo/paint_bezier.rs

```diff
@@ -0,0 +1,182 @@
+                if ui.button("Clear Painting").clicked() {
+                    self.points.clear();
+                    self.backup_points.clear();
+                    self.quadratic_shapes.clear();
+                    self.cubic_shapes.clear();
+                }
+                ui.label("Click 3 or 4 points to build a bezier curve!");
+            })
+            
+        })
+        .response
+    }
+
+    // an internal function to create auxiliary lines around the current bezier curve
+    // or to auxiliary lines (points) before the points meet the bezier curve requirements.
+    fn build_auxiliary_line(&self,points:&[Pos2],to_screen:&RectTransform,aux_stroke:&Stroke)->Vec<Shape>{
+        let mut shapes = Vec::new();
+        if points.len()>=2 {
+            let points:Vec<Pos2> = points.iter().map(|p| to_screen * *p).collect();
+            shapes.push(egui::Shape::line(points, aux_stroke.clone()));
+        }
+        for point in points.iter(){
+            let center = to_screen * *point;
+            let radius = aux_stroke.width * 3.0;
+            let circle = CircleShape {
+                center,
+                radius,
+                fill: aux_stroke.color,
+                stroke: aux_stroke.clone(),
+            };
+            
+            shapes.push(circle.into());
+        }
+        
+        shapes
+    }
+
+    pub fn ui_content(&mut self, ui: &mut Ui) -> egui::Response {
+        let (mut response, painter) =
+            ui.allocate_painter(ui.available_size_before_wrap(), Sense::click());
+
+        let to_screen = emath::RectTransform::from_to(
+            Rect::from_min_size(Pos2::ZERO, response.rect.square_proportions()),
+            response.rect,
+        );
+        let from_screen = to_screen.inverse();
+
+        if response.clicked() {
+            if let Some(pointer_pos) = response.interact_pointer_pos() {
+                let canvas_pos = from_screen * pointer_pos;
+                self.points.push(canvas_pos);
+                if self.points.len() >= self.bezier {
+                    self.backup_points = self.points.clone();
+                    let points = self.points.drain(..).collect::<Vec<_>>();
+                    match points.len(){
+                        3 => {
+                            let points: Vec<Pos2> = points.iter().map(|p| to_screen * *p).collect();
+                            let stroke = self.stroke.clone();
+                            let quadratic = QuadraticBezierShape::from_points_stroke(points, stroke);
+                            self.quadratic_shapes.push(quadratic);
+                        },
+                        4 => {
+                            let points: Vec<Pos2> = points.iter().map(|p| to_screen * *p).collect();
+                            let stroke = self.stroke.clone();
+                            let cubic = CubicBezierShape::from_points_stroke(points, stroke);
+                            self.cubic_shapes.push(cubic);
```

Comment:
It would be nice to add a checkbox to the demo where you can also fill the bezier curves (since the they have the `closed + fill` members

Reply:
yes, added a checkbox for `closed` and a color picker for `fill`. Also, I added a new logic to check whether the curve will cross the baseline or not. The default `fill` doesn't work as expected when it's not convex.